src/v/raft/consensus.cc

```diff
@@ -2468,15 +2468,17 @@ ss::future<> consensus::maybe_commit_configuration(ssx::semaphore_units u) {
     /**
      * current config still contains learners, do nothing
      */
-    if (unlikely(!latest_cfg.current_config().learners.empty())) {
-        co_return;
-    }
     switch (latest_cfg.get_state()) {
     case configuration_state::simple:
         co_return;
     case configuration_state::transitional:
         vlog(
           _ctxlog.info, "finishing transition of configuration {}", latest_cfg);
+
+        if (unlikely(!latest_cfg.current_config().learners.empty())) {
+            co_return;
+        }
+
```

Comment:
nit: It isn't really clear to me (at least without going through more code) what the nuances between simple, transitional, and joint are for this function. Might be worth adding a comment, at least a conceptual one, so it's more obvious for readers that this is the right place for this line of code (and so it's obvious they shouldn't move it back accidentally!). Ideally the comment isn't specific to this code change, and more general though.

Reply:
Alternatively, pointers to important context would be appreciated.

Reply:
added